cloudflare/resource_cloudflare_notification_policy_webhooks.go

```diff
@@ -0,0 +1,175 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceCloudflareNotificationPolicyWebhooks() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareNotificationPolicyWebhooksCreate,
+		Read:   resourceCloudflareNotificationPolicyWebhooksRead,
+		Update: resourceCloudflareNotificationPolicyWebhooksUpdate,
+		Delete: resourceCloudflareNotificationPolicyWebhooksDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareNotificationPolicyWebhooksImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"secret": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_success": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_failure": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareNotificationPolicyWebhooksCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get("account_id").(string)
+
+	notificationWebhooks := buildNotificationPolicyWebhooks(d)
+
+	webhooksDestination, err := client.CreateNotificationWebhooks(context.Background(), accountID, &notificationWebhooks)
+
+	if err != nil {
+		return fmt.Errorf("error connecting webhooks destination %s: %s", notificationWebhooks.Name, err)
+	}
+	formattedWebhookID, err := uuid.Parse(webhooksDestination.Result.ID)
```

Comment:
i'm not sure we really need this. what is the intention over just using `webhooksDestination.Result.ID` as the `d.SetId()` value?

Reply:
the notifications API checks if the webhook id received is a valid UUID or not. When we POST a webhook, the ID returned is not formatted with dashes `-` (although it's a valid UUID).  But when we update a webhook that's how it's expected.

Reply:
When a webhook is created an ID is returned, which is a UUID. This may or may not have the dashes. But when we update a webhook we check if it's in the right format with dashes, so this is just an additional check to make sure they're formatted correctly.

Reply:
Thanks for the info. We may want to address this service side in the future to return a consistent format as logic/safeguards like this in the Terraform provider will need to be duplicated into `cf-terraforming`. It's also not a great experience if we have potentially inconsistent identifier for resources. 

Let's proceed with this for now and discuss any improvements later on.

Reply:
Agree with what you're saying. I'll create a ticket to fix it in our service and update here later on.